src/adapter/src/catalog/migrate.rs

```diff
@@ -588,3 +591,17 @@ fn catalog_remove_txn_wal_toggle_v_0_109_0(tx: &mut Transaction) -> Result<(), a
     tx.remove_system_config("persist_txn_tables");
     Ok(())
 }
+
+/// This migration will purge pending replicas from the catalog. These replicas
+/// were part of an uncompleted cluster reconfiguration and will be cleaned up
+/// by `remove_orphaned_replicas`.
+fn catalog_remove_pending_replicas(tx: &mut Transaction) -> Result<(), anyhow::Error> {
+    for replica in tx.get_cluster_replicas() {
+        if let mz_catalog::durable::ReplicaLocation::Managed { pending: true, .. } =
+            replica.config.location
+        {
+            tx.remove_cluster_replica(replica.replica_id)?;
+        }
+    }
+    Ok(())
+}
```

Comment:
Honestly, this feels a little out of place here. It feels more appropriate as a block in `initialize_state` in `open.rs`.

Reply:
Yeah, that does seem better! It's been moved.